agent/api/task/taskvolume_test.go

```diff
@@ -241,14 +245,63 @@ func TestInitializeSharedNonProvisionedVolume(t *testing.T) {
 			Labels: map[string]string{"test": "test"},
 		},
 	})
-	err := testTask.initializeDockerVolumes(dockerClient, nil)
+	err := testTask.initializeDockerVolumes(&cfg, dockerClient, nil)
+
+	assert.NoError(t, err)
+	assert.Len(t, testTask.ResourcesMapUnsafe, 0, "no volume resource should be provisioned by agent")
+	assert.Len(t, testTask.Containers[0].TransitionDependenciesMap, 0, "resource already exists")
+}
+
+func TestInitializeSharedNonProvisionedVolumeValidateNameOnly(t *testing.T) {
+	cfg := config.Config{}
+	cfg.SharedVolumeValidateNameOnly = true
+
+	ctrl := gomock.NewController(t)
```

Comment:
i think you need to have 'defer ctrl.Finish()' after that so the EXPECTed call will be checked?

Reply:
added these on a separate commit